### PR TITLE
v2.12.0 (issue210-address-as-parameter) - Adding ADDRESS as a potential parameter

### PR DESCRIPTION
**Pull request by trevorrd**
This PR is to address issue210-address-as-parameter. I've made the changes to now allow developers to use an address instead of latitude and longitude coordinates. This is what I had commented on earlier and you, Thomas Schoffelen, had commented and said, `It's not a use case I personally have, because I want to send users to a specific location rather than doing a search, but I can understand why that would be useful. I'm still happy to accept a PR for that functionality if you'd like to add it!`

So here is my pull request. Please review it, give me any feedback, if you see anything I need to update or change please let me know.

### CODE REVIEW

**Comment by trevorrd** [comment]
I just noticed you have another PR that is going to convert this library into TypeScript (v3.0.0). If you want to do that first then we can go in and make similar changes we can do that too.

**Comment by tschoffelen** [comment]
Hi @trevorrd - apologies for the delay! The Typescript migration has now been completed, so if you're willing to update this, I'm happy to get this merged. Apologies for the duplicated work!

**Comment by trevorrd** [comment]
> Hi @trevorrd - apologies for the delay! The Typescript migration has now been completed, so if you're willing to update this, I'm happy to get this merged. Apologies for the duplicated work!

Hey Thomas,
I've made the changes and pushed them up. This PR should be ready for a review. I've also been pretty busy in the last few weeks but I still feel these changes would be very beneficial. 

Thanks!

File: src/type.ts [hunk 1]
```diff
@@ -47,6 +47,9 @@ export type GetAppsResponse = {
 export interface ShowLocationProps {
   latitude: number | string;
```
Comment (jerone) [inline src/type.ts:48]: I'm interested in the address property only for my use-case. 
Should `latitude` & `longitude` be made optional?
```suggestion
  latitude?: number | string;
  longitude?: number | string;
```

Comment (trevorrd) [reply src/type.ts]: jerone, I've made the change. If you see anything else, please let me know.

Comment (jerone) [reply src/type.ts]: The example still shows empty strings.

Comment (trevorrd) [reply src/type.ts]: Thank you. Modified it.

File: README.md [hunk 2]
```diff
@@ -265,12 +265,31 @@ showLocation({
 });
 ```
 
+Alternatively you can specify the `address` field and leave the latitude and longitude properties as empty strings 
+
+```js
+import {showLocation} from 'react-native-map-link';
+
+showLocation({
```
Comment (tschoffelen) [inline README.md:273]: suggestion: could we do a simpler example here, without all of the optional fields, to better illustrate this alternative usage?

**Review by tschoffelen — COMMENTED** [review]
This is amazing work! One quick suggestion on the README, but looking very much forward to merging this soon!

Thanks again!!

File: README.md (hunk 2 above)
Comment (trevorrd) [reply README.md:273]: Awesome. I made that change. 👍

**Review by tschoffelen — APPROVED** [review]

**Comment by tschoffelen** [comment]
It's taken me way too long (apologies again), but it's merged! Thanks again @trevorrd for the amazing work on this, quite a big new feature you've shipped here!

**Comment by OwenMelbz** [comment]
Hi,

Is anybody able to confirm this works with google maps on ios?

Every time we use the address field it, then launch maps it gets "google maps cant open this link"

image

IMG_9A05AED0C74A-1

**Comment by OwenMelbz** [comment]
Update:

After some debugging it looks like the url structure is wrong.

Referring to the google docs here: https://developers.google.com/maps/documentation/urls/get-started it says all urls should have `api=1` in, which we can see is used later on:

image

However for the address line it instead uses `?q=$address`, whereas the docs about says it should be `?api=1&query=$address` upon changing that, it worked, so i believe there is a bug there

**Comment by trevorrd** [comment]
> Update:
> 
> After some debugging it looks like the url structure is wrong.
> 
> Referring to the google docs here: https://developers.google.com/maps/documentation/urls/get-started it says all urls should have `api=1` in, which we can see is used later on:
> 
> image
> However for the address line it instead uses `?q=$address`
> 
> image
> whereas the docs about says it should be `?api=1&query=$address` upon changing that, it worked, so i believe there is a bug there
> 
> image
> PR to fix is here: #294
> 
> Would be great to get this merged asap if possible as we caught this issue during testing for a big app update which meant to be going out today :D

Nice catch! Yes, open a bug ticket. I'm not sure how I missed that small query string parameter. I know I used it in my project for work and must have missed it as I was adding it to this library. If we could get the ticket opened and change that, it would be great. Thanks!

**Comment by OwenMelbz** [comment]
There's already a PR up for it here: https://github.com/includable/react-native-map-link/pull/294

**Pull request merged**